api/handlers/app_test.go

```diff
@@ -1772,6 +1774,113 @@ var _ = Describe("App", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /v3/apps/:guid/processes/:process/instances/:instance", func() {
```

Comment:
Maybe a better structure for this describe would be 

```
Describe("DELETE /v3/apps/:guid/processes/:process/instances/:instance", func() {
  It("returns process not found error", func() {})

  When("the app does not exist", func() {
    It("returns an app not found error", func() {})
  })

  When("the app has a process", func() {
    It("returns an instance not found error", func() {})

    When("the process has an instance") {
      It("restarts the instance", func() {})
    })

    When("the app has more than one process", func() {
      It("restarts the instance", func() {})
    })
  })
})
```

The idea is that each when builds on top of its parent. This way a lot of repetition can be avoided